lib/spack/spack/spec.py

```diff
@@ -1012,9 +1012,14 @@ def __init__(self, spec_like, *dep_like, **kwargs):
         self._hash = other._hash
         self._cmp_key_cache = other._cmp_key_cache
 
-        # Specs are by default not assumed to be normal or concrete.
-        self._normal = False
-        self._concrete = False
+        self._prefix = getattr(other, '_prefix', None)
+
+        # Specs are by default not assumed to be normal, but in some
+        # cases we've read them from a file want to assume normal.
+        # This allows us to manipulate specs that Spack doesn't have
+        # package.py files for.
+        self._normal = kwargs.get('normal', False)
```

Comment:
@alalazo You changed these two lines to be 

```
self._normal = False
self._concrete = False
```

in #4129. I don't see why they had to be changed, so I left them as is when I did the merge, but let me know if that will cause problems in something I'm not fully understanding with mv variants.